### PR TITLE
Remove PersistentWorkflowInstanceState class

### PR DESCRIPTION
**Pull request by fabriziodemaria**
After #385 `PersistentWorkflowInstanceState` variables aligned with `RunState` variables, making the former class not needed anymore. The PR also renames some related method for storing/fetching active states for better naming consistency. 

This also removes some TODOs that remove code related to the transition of active states into the persistent layer.

### CODE REVIEW

File: styx-scheduler-service/src/main/java/com/spotify/styx/state/QueuedStateManager.java [hunk 1]
```diff
@@ -200,13 +192,11 @@ private void initialize(WorkflowInstance workflowInstance) {
...
         // Write new state to datastore (or remove it if terminal)
-        final long nextCounter = persistentState.get().counter() + 1;
+        final long nextCounter = currentRunState.get().counter() + 1;
         if (nextRunState.state().isTerminal()) {
           tx.deleteActiveState(event.workflowInstance());
         } else {
-          final PersistentWorkflowInstanceState nextPersistentState =
-              PersistentWorkflowInstanceState.of(nextRunState, nextCounter);
-          tx.updateActiveState(event.workflowInstance(), nextPersistentState);
+          tx.updateActiveState(event.workflowInstance(), nextRunState);
```
Comment (NatashaL) [inline styx-scheduler-service/src/main/java/com/spotify/styx/state/QueuedStateManager.java]: `nextCounter` is lost here, should be set in `nextRunState`

Comment (fabriziodemaria) [reply styx-scheduler-service/src/main/java/com/spotify/styx/state/QueuedStateManager.java]: I actually moved the counter increasing within the `transition` method in `RunState`, so it's done "automatically".

**Comment by NatashaL** [comment]
👍

**Comment by honnix** [comment]
Put some description to be PR? It’s hard to follow the reasoning.

**Comment by fabriziodemaria** [comment]
> Put some description to be PR? It’s hard to follow the reasoning.

Added 
@honnix

**Comment by honnix** [comment]
Cool. Please feel free to merge. You got +1 already.
